lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -47,6 +47,10 @@ type AddInvoiceConfig struct {
 	// ChanDB is a global boltdb instance which is needed to access the
 	// channel graph.
 	ChanDB *channeldb.DB
+
+	// GenInvoiceFeatures returns a feature containing feature bits that
+	// should be advertised on freshly generated invoices.
+	GenInvoiceFeatures func() *lnwire.FeatureVector
```

Comment:
Any reason this can't just be a field rather than a function closure?

Reply:
i wanted to make sure that there was no possibility of mutation, so this allows each invoice to have a distinct feature vector. we could just used the `Clone()` method, but in the future it's likely that this set will be dynamic depending on the type of invoice being generated, so i thought it made sense to make it a function